catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/QueryResultIterable.java

```diff
@@ -131,9 +136,17 @@ private void queryNext(int index) {
                 Map<String, Serializable> props = new HashMap<>();
                 // Avoid caching all results while dumping with native query mode
                 props.put("mode", "native");
-                response = catalog.query(filter.apply(index));
+                response = catalog.query(new QueryRequestImpl(new QueryImpl(filter,
+                        pageIndex,
+                        pageSize,
+                        SortBy.NATURAL_ORDER,
+                        false,
+                        5000l/*timeout? config? passed in?*/)));
+                missingResults = pageSize - response.getResults()
+                        .size();
+
             } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
-                throw new CatalogCommandRuntimeException(e);
+                throw new CatalogRuntimeException(e);
```

Comment:
Certainly. At least for the errors that pop up from the catalog framework. Would much rather not hide those unless anyone has strong feelings for hiding them. Especially if we will already have to modify code to replace instances with this, we should also make sure we properly deal with exceptions.